common/djangoapps/track/views/tests/test_views.py

```diff
@@ -321,18 +328,17 @@ def test_task_track(self):
         views.task_track(request_info, task_info, str(sentinel.event_type), self.event)
 
         expected_event = {
-            'username': 'anonymous',
-            'ip': '127.0.0.1',
-            'event_source': 'task',
-            'event_type': str(sentinel.event_type),
-            'event': expected_event_data,
-            'agent': 'agent',
-            'page': None,
-            'time': FROZEN_TIME,
-            'host': 'testserver',
+            'timestamp': FROZEN_TIME,
+            'data': expected_event_data,
+            'name': str(sentinel.event_type),
             'context': {
                 'course_id': '',
-                'org_id': ''
-            },
+                'username': 'anonymous',
+                'org_id': '',
+                'page': None,
+                'event_source': 'task'
+            }
```

Comment:
To make sure I understand the repercussions of this change, the event output will now be different moving forward, correct?  Can you summarize the event schema changes?  For example:

* `event` -> `data`
* fields moved to `context`

I am trying to determine whether this will result in backward compatibility issues for event researchers.

Reply:
@nasthagiri It is not going to change the event output much because we already have [LegacyFieldMappingProcessor](https://github.com/edx/edx-platform/blob/046feb0cf0c02acb562739bb9128755755ef3157/common/djangoapps/track/shim.py#L19) in place. which is responsible for mapping `eventracking` event fields to `track` event fields.
I'm attchaching comparasion of fields in `eventracking` event and `track` event.  `eventracking` event has two extra fields `name` and `session`.
![screencapture-2020-06-24-17_05_31](https://user-images.githubusercontent.com/434995/85558528-40d4b500-b642-11ea-9829-2a09d163e76b.png)

Reply:
Awesome! In that case, this looks good to me.

@macdiesel @mulby @doctoryes I am planning to merge this at noon EST tomorrow (Thurs). Do you have any concerns from your end?

Reply:
We need to make sure context.course_id is properly set for the new events as this determines the routing of the events to researchers. I'll put a note on the line that I think needs to change.